app/src/main/java/de/tum/in/tumcampusapp/component/ui/search/adapter/ResultTypesAdapter.kt

```diff
@@ -42,49 +39,65 @@ class ResultTypesAdapter(
 
         fun bind(typeData: ResultTypeData) {
             currentType = typeData
-            val text = when (typeData.type) {
+            val textId = when (typeData.type) {
                 SearchResultType.ALL -> R.string.all_results
                 SearchResultType.PERSON -> R.string.people
                 SearchResultType.LECTURE -> R.string.lectures
                 SearchResultType.NAVIGA_ROOM -> R.string.rooms_id
                 SearchResultType.BUILDING -> R.string.buildings
             }
-            typeTextView.setText(text)
 
-            if (typeData.type == typeData.selectedType) {
-                typeTextView.textColorResource = R.color.white
-                layout.setBackgroundResource(R.drawable.search_result_selected_type_background)
+            chip.text = chip.context.resources.getText(textId)
+            setIsSelected(typeData.selected)
+
+            if (typeData.type == SearchResultType.ALL) {
+                itemView.margin(left = 16F)
             } else {
-                typeTextView.textColorResource = R.color.text_primary
-                layout.setBackgroundResource(R.drawable.search_result_type_background)
+                itemView.margin(left = 8F)
             }
```

Comment:
I dont fully know about this part (this is fine too)

what do you think about this alternative?
````kotlin
val leftMargin = when (typeData.type == SearchResultType.ALL) {
                true -> 16F
                false -> 8F
            };
            itemView.margin(left = leftMargin)

Reply:
I have added an even prettier variant using the xml-attribute `clipToPadding`, removing this logic entirely.

Reply:
Looks way better. Agree